### PR TITLE
JNG-4767 JSL UI syntax

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4767" title="JNG-4767" target="_blank">StoryJNG-4767</a>  JSL UI syntax
  </td></table>
  <br />
 

JNG-4767 JSL UI syntax

### CODE REVIEW

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityIdentifierDeclaration.eol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@cached
-operation JSL!EntityIdentifierDeclaration getId(): String {
-    return (self.eContainer.getId() + "/" + self.name);
-}
+//@cached
+//operation JSL!EntityIdentifierDeclaration getId(): String {
+//    return (self.eContainer.getId() + "/" + self.name);
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityIdentifierDeclaration.eol]: Dead code. Is identifier exists?

Comment (robertcsakany) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityIdentifierDeclaration.eol]: And please remove EOL file if empty

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityQueryDeclaration.eol [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@cached
-operation JSL!EntityQueryDeclaration getId(): String {
-    return (self.eContainer.getId() + "/" + self.name);
-}
+//@cached
+//operation JSL!EntityQueryDeclaration getId(): String {
+//    return (self.eContainer.getId() + "/" + self.name);
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityQueryDeclaration.eol]: Please remove deasd code

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityIdentifierDeclaration.eol (hunk 1 above)
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityIdentifierDeclaration.eol]: Please remove dead code and the file

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/primitiveTypedElement.etl [hunk 3]
```diff
@@ -10,28 +10,30 @@ rule CreatePrimitiveTypedElement
...
+//    to t: JUDOPSM!Attribute
+//    extends CreatePrimitiveTypedElement {
+//        t.setId("(jsl/" + s.getId() + ")/CreateAttributeFromIdentifier");
+//
+//        t.dataType = s.getReferenceType().getPSMEquivalent();
+//        t.identifier = true;
+//        s.eContainer.getPSMEquivalent().attributes.add(t);
+//        log.debug("Created Attribute (Identifier): " + t.name);
+//    }
+//
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/primitiveTypedElement.etl]: Please remove dead code

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationAutoMappedTransferAttribute.etl [hunk 4]
```diff
@@ -48,43 +48,43 @@ rule CloneTransferAttributeForAutoMappedTransferObjectType
...
+//
+//@lazy
+//@greedy
+//rule CloneIdentifierTransferAttributeForAutoMappedTransferObjectType
+//    transform s: JSL!EntityIdentifierDeclaration
+//    to t: JUDOPSM!TransferAttribute
+//    extends AbstractCreateIdentifierTransferAttributeForAutoMappedTransferObjectType {
+//        t.setId("(jsl/" + s.getId() + ")/CloneIdentifierTransferAttributeForAutoMappedTransferObjectType");
+//        log.debug("Clone TransferAttribute (Identifier) for AutoMappedTransferObjectType: [" + t.name + "]");
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationAutoMappedTransferAttribute.etl]: Please remove dead code

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationAutoMappedTransferObjectType.etl [hunk 5]
```diff
@@ -27,26 +27,26 @@ rule CreateAutoMappedTransferObjectType
 
             for (im in inheritedMembers) {
                 // Identifier attribute
-                if (im.isKindOf(JSL!EntityIdentifierDeclaration)) {
-                    t.attributes.add(im.equivalentWithPrefix("CloneIdentifierTransferAttributeForAutoMappedTransferObjectType", t.getId()));
+//                if (im.isKindOf(JSL!EntityIdentifierDeclaration)) {
+//                    t.attributes.add(im.equivalentWithPrefix("CloneIdentifierTransferAttributeForAutoMappedTransferObjectType", t.getId()));
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationAutoMappedTransferObjectType.etl]: Please remove dead code

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationDefaultTransferAttribute.etl [hunk 6]
```diff
@@ -50,44 +50,44 @@ rule CloneTransferAttributeForDefaultTransferObjectType
...
+//
+//@lazy
+//@greedy
+//rule CloneIdentifierTransferAttributeForDefaultTransferObjectType
+//    transform s: JSL!EntityIdentifierDeclaration
+//    to t: JUDOPSM!TransferAttribute
+//    extends AbstractCreateIdentifierTransferAttributeForDefaultTransferObjectType {
+//        t.setId("(jsl/" + s.getId() + ")/CloneIdentifierTransferAttributeForDefaultTransferObjectType");
+//        log.debug("Clone TransferAttribute (Identifier) for DefaultTransferObjectType: [" + t.name + "]");
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/entityDeclarationDefaultTransferAttribute.etl]: Please remove dead code

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl [hunk 7]
```diff
@@ -60,20 +60,22 @@ rule CreateMappedTransferAttribute
...
-         t.binding = s.maps.features.first.member.equivalent("CreateAttributeFromIdentifier");
+      if (s.getterExpr.features.first.member.isKindOf(JSL!EntityStoredFieldDeclaration)) {
+         t.binding = s.getterExpr.features.first.member.equivalent("CreateAttributeFromField");         
       }
       
-      var def = JSL!TransferDefault.all().selectOne(d | d.field.reference == s);
+//      else if (s.maps.features.first.member.isKindOf(JSL!EntityIdentifierDeclaration)) {
+//         t.binding = s.maps.features.first.member.equivalent("CreateAttributeFromIdentifier");
+//      }
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: Please remove dead code

**Review by robertcsakany — CHANGES_REQUESTED** [review]

**Review by gaborflorian — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Pull request merged**